state/execution_test.go

```diff
@@ -31,6 +35,32 @@ func TestExecBlock(t *testing.T) {
 	// TODO
 }
 
+func TestApplyBlock(t *testing.T) {
```

Comment:
Great we need some tests here. There was an earlier attempt to test ExecBlock (https://github.com/tendermint/tendermint/pull/328/files) but it's more complicated than it needs to be since it relies on event subscriptions - but now we can access the tx results directly!

Also, sorry, but I totally gutted this file on develop. All the handshake stuff moved to consensus/replay_test.go. So rebasing might be a bit painful around here